base/compiler/typeinfer.jl

```diff
@@ -387,6 +387,11 @@ function cache_result!(interp::AbstractInterpreter, result::InferenceResult)
         relocatability = isa(inferred_result, Vector{UInt8}) ? inferred_result[end] : UInt8(0)
         code_cache(interp)[linfo] = CodeInstance(result, inferred_result, valid_worlds, relocatability)
     end
+
+    if isdefined(result, :escapes)
+        EscapeAnalysis.GLOBAL_ESCAPE_CACHE[linfo]= result.escapes
```

Comment:
@ianatol we want to modify this part so that we directly keep `result.escapes` within the `CodeInstance` created above.